src/Nancy.ViewEngines.Razor/ClrTypeResolver.cs

```diff
@@ -0,0 +1,131 @@
+﻿namespace Nancy.ViewEngines.Razor
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Text;
+    using System.Web.Razor.Tokenizer.Symbols;
+
+    using Nancy.Bootstrapper;
+
+    /// <summary>
+    /// Converts language-specific type name into CRL Type
+    /// </summary>
+    /// <typeparam name="TSymbolType">Symbol type enum</typeparam>
+    /// <typeparam name="TSymbol">Symbol class</typeparam>
+    internal abstract class ClrTypeResolver<TSymbolType, TSymbol>

```

Comment:
1. `public` / `protected` members should have XML comments
2. The methods of this class could do with a bit of refactoring (being split into helper methods) to make it easier to follow the logic